common/lib/protocol-definitions/src/protocol.ts

```diff
@@ -312,6 +312,17 @@ export interface ISignalMessage {
 	// eslint-disable-next-line @rushstack/no-new-null
 	clientId: string | null;
 
+	/**
+	* The source of the signal. This can be a GUID of the client,
```

Comment:
This is great documentation! One nitpick recommendation: TSDoc encourages putting semantic overview documentation in the core "summary" (no tag) part of the comment. But usage notes, including the note you added about backwards compatibility are usually better places in a `@remarks` block.